### PR TITLE
React: hook up analysis table run / cancel analyses

### LINKED ISSUE #80: Hook up analysis page mutations to PATCH endpoints
Frontend for mutations from #46, #47, #49 were merged in #62
Backend endpoint merged in #37

- [x] Table rows
- [x] Set assignee
- [ ] Cancel analysis
- [ ] Run/finish analysis

### PR DESCRIPTION
**Pull request by frewmack**
Closes #80

### CODE REVIEW

**Review by kevinlul — COMMENTED** [review]
This looks great! Is the entire scope of the pull request just the two existing buttons? As long as we don't have automated pipeline runs, we'll need buttons to manually set the state to running and completed. Cancelled should be a dead state that doesn't get restarted.

**Comment by frewmack** [comment]
> 
> 
> This looks great! Is the entire scope of the pull request just the two existing buttons? As long as we don't have automated pipeline runs, we'll need buttons to manually set the state to running and completed. Cancelled should be a dead state that doesn't get restarted.

I must've misunderstood the requirements then. Should I add two more buttons for setting state to running and completed, and rename the existing 'run' button to something like 'request analysis'?

**Comment by kevinlul** [comment]
- [ ] Rename PENDING on frontend to REQUESTED to match the backend
- [ ] Cancel button. No cancelling completed or errored analyses
- [ ] Errored button. No erroring completed or cancelled analyses
- [ ] Run button. Can only be applied to requested analyses
- [ ] Complete button. Can only be applied to running analyses

Hope this helps!

File: react/src/pages/analysis/Analyses.tsx [hunk 1]
```diff
@@ -349,24 +409,54 @@ export default function Analyses() {
...
-                                enqueueSnackbar(`${count} ${count !== 1 ? 'analyses' : 'analysis'} started successfully`);
                             },
                         },
+                        {
+                            icon: AssignmentTurnedIn,
+                            tooltip: 'Complete analysis',
+                            position: 'toolbarOnSelect',
+                            onClick: (event, rowData) => {
+                                setActiveRows(rowData as Analysis[]);
+                                changeStateForSelectedRows(rows, completeFilter, PipelineStatus.COMPLETED)
```
Comment (kevinlul) [inline react/src/pages/analysis/Analyses.tsx:430]: The state in the backend is actually named "Done"

File: react/src/pages/analysis/Analyses.tsx [hunk 2]
```diff
@@ -138,18 +137,80 @@ function rowsToString(rows: Analysis[], delim?: string) {
...
-    return row.state === PipelineStatus.RUNNING || row.state === PipelineStatus.PENDING;
+    return row.analysis_state === PipelineStatus.RUNNING || row.analysis_state === PipelineStatus.PENDING;
 }
 
 /**
  * Returns whether this analysis is allowed to be run.
  */
 function runFilter(row: Analysis) {
-    return row.state === PipelineStatus.ERROR || row.state === PipelineStatus.CANCELLED;
+    return row.analysis_state === PipelineStatus.ERROR || row.analysis_state === PipelineStatus.COMPLETED;
```
Comment (kevinlul) [inline react/src/pages/analysis/Analyses.tsx:165]: I believe this logic is inverted. Only requested analyses can be run.

**Review by kevinlul — CHANGES_REQUESTED** [review]

**Review by kevinlul — APPROVED** [review]
LGTM

**Pull request merged**